community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/pipes/ShortestPathPipe.scala

```diff
@@ -66,10 +66,8 @@ case class ShortestPathPipe(source: Pipe, shortestPathCommand: ShortestPath, pre
   }
 
   override def planDescriptionWithoutCardinality = {
-    val args = predicates.map { p =>
-      LegacyExpression(p)
-    }
-    source.planDescription.andThen(this.id, "ShortestPath", variables, args:_*)
+    val args = predicates.zipWithIndex.map { case (p, idx) => s"p$idx" -> p }
```

Comment:
in this case - predicate. An alternative could be to AND the predicates together and show the ANDed tree instead. Do you think that would be a better solution?